File: 02. Algorithms/03. Searching/01. Linear Search/Linear_Search.java
// TODO: Search an Element in the array using Linear Search Technique
package array.searching;

public class Linear_Search {

	static int linearSearch(int[] arr, int element) {
		if(arr.length == 0)
			return -1;
		
		for(int i = 0; i < arr.length; i++)
		{
			// find the first occurrence of the element and return it's position
			if(arr[i] == element) {
				return i + 1;
			}
		}
		return -1;
	}
	public static void main(String[] args) {
		int[] arr = { 18, 12, 9, 14, 77, 50, 45, 2, 91, 40};
		int element = 9;
		int res = linearSearch(arr, element);
		if(res == -1)
			System.out.println("Element not Found");
		else
			System.out.println("Element Found at position " + res);
	}
}

File: 02. Algorithms/03. Searching/01. Linear Search/Linear_Search.cpp
#include<iostream>
using namespace std;
void linear(int a[], int n) {
  int t = -1;
for (int i = 0; i < 5; i++) {
    if (a[i] == n) {
      cout << "Element found at position: " << i + 1 << endl;
      t = 0;
      break;
    }
  }
 if (t == -1) {
    cout << "No Element Found" << endl;
  }}
int main() {
  int arr[5];
  cout << "Please enter 5 elements of the Array" << endl;
  for (int i = 0; i < 5; i++) {
    cin >> arr[i];
  }
  cout << "Please enter an element to search" << endl;
  int num;
  cin >> num;
linear(arr, num);
  return 0;
}
